django/test/signals.py

```diff
@@ -172,3 +173,24 @@ def auth_password_validators_changed(**kwargs):
 def user_model_swapped(**kwargs):
     if kwargs['setting'] == 'AUTH_USER_MODEL':
         apps.clear_cache()
+        try:
+            from django.contrib.auth import get_user_model
+            UserModel = get_user_model()
+        except ImproperlyConfigured:
+            # Some tests set an invalid AUTH_USER_MODEL.
+            pass
+        else:
+            from django.contrib.auth import backends
```

Comment:
Is it worth documenting? Maybe it's worth keeping documentation about referencing `settings.AUTH_USER_MODEL` to avoid that?